src/lints/inherent_associated_pub_const_missing.ron

```diff
@@ -0,0 +1,64 @@
+SemverQuery(
+    id: "inherent_associated_pub_const_missing",
+    human_readable_name: "inherent impl's associated pub const removed",
+    description: "An inherent impl's associated public const removed or renamed",
+    required_update: Major,
+    reference_link: Some("https://doc.rust-lang.org/cargo/reference/semver.html#item-remove"),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on ImplOwner {
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+
+                        importable_path {
+                            path @output @tag
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        inherent_impl {
+                            public_api_eligible @filter(op: "=", value: ["$true"])
+                            
+                            associated_constant {
+                                associated_constant: name @output @tag
+                                public_api_eligible @filter(op: "=", value: ["$true"])
+
+                                span_: span @optional {
+                                    filename @output
+                                    begin_line @output
+                                }
+                            }    
+                        }
+                    }
+                }
+            }
+            current {
+                item {
+                    ... on ImplOwner {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        name @output
+
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        inherent_impl @fold @transform(op: "count") @filter(op: "=", value: ["$zero"]){
```

Comment:
Ah one last test case is needed here.

The following might not always be a breaking change, and should be covered by a separate lint:
```rust
pub trait Trait {}

pub struct Example;

impl Example {
    pub const N: i64 = 0;
}

impl Trait for Example {}
```
to
```rust
pub trait Trait {
    const N: i64 = 0;
}

pub struct Example;

impl Trait for Example {}
```

The reason this might not be breaking is that if both `Example` and `Trait` are in a prelude, or if the `Trait` trait is already otherwise in scope, `Example::N` will continue to work. In the original code that refers to an inherent const, but in the new code it's a trait's const, equivalent to `<Example as Trait>::N` in explicit form.

To fix this, we allow the query to look at _all_ `impl` blocks (including `impl Trait for Example` ones) and not just inherent `impl` (only `impl Example`) blocks:
```suggestion
                        impl @fold @transform(op: "count") @filter(op: "=", value: ["$zero"]){
```

It would be great if you could add the above code as a test case here.